### ISSUE TITLE
Build failure: pw-viz

### ISSUE THREAD

**Issue by toastal** [NixOS/nixpkgs#268450]
### Steps To Reproduce

1. build `pw-viz`

### Build log

https://paste.sr.ht/~toastal/4a0052055f65bc36ea30b113cae20a152ae52986

### Notify maintainers
@figsoda

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix shell nixpkgs#nix-info --command nix-info -m                                                                                                                          
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.11, NixOS, 23.11 (Tapir), 23.11.20231117.c757e9b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(toastal): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos
```

**Comment by chewblacka** [comment]
```
error: failed to run custom build command for `libspa-sys v0.5.0`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/libspa-sys-eb834811139a49bb/build-script-build` (exit status: 101)

  --- stderr
  thread 'main' panicked at /build/cargo-vendor-dir/proc-macro2-1.0.49/src/fallback.rs:756:9:
  "spa_audio_info_union_(unnamed_at_/nix/store/cc1lqllsa8yfl1jvnyl4wbc0ciik2yfr-pipewire-0_3_84-dev/include/spa-0_2/spa/param/audio/format_h_36_2)" is not a valid Ident
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: builder for '/nix/store/75h0q77r8hc3wi4kqpg2lvnjx9wvb4nq-pw-viz-0.2.0.drv' failed with exit code 101;
       last 10 log lines:
       >   cargo:rerun-if-changed=/nix/store/cc1lqllsa8yfl1jvnyl4wbc0ciik2yfr-pipewire-0.3.84-dev/include/spa-0.2/spa/pod/iter.h
       >   cargo:rerun-if-changed=/nix/store/cc1lqllsa8yfl1jvnyl4wbc0ciik2yfr-pipewire-0.3.84-dev/include/spa-0.2/spa/pod/parser.h
       >   cargo:rerun-if-changed=/nix/store/cc1lqllsa8yfl1jvnyl4wbc0ciik2yfr-pipewire-0.3.84-dev/include/spa-0.2/spa/pod/pod.h
       >   cargo:rerun-if-changed=/nix/store/cc1lqllsa8yfl1jvnyl4wbc0ciik2yfr-pipewire-0.3.84-dev/include/spa-0.2/spa/pod/vararg.h
       >
       >   --- stderr
       >   thread 'main' panicked at /build/cargo-vendor-dir/proc-macro2-1.0.49/src/fallback.rs:756:9:
       >   "spa_audio_info_union_(unnamed_at_/nix/store/cc1lqllsa8yfl1jvnyl4wbc0ciik2yfr-pipewire-0_3_84-dev/include/spa-0_2/spa/param/audio/format_h_36_2)" is not a valid Ident
       >   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
       > warning: build failed, waiting for other jobs to finish...
       For full logs, run 'nix log /nix/store/75h0q77r8hc3wi4kqpg2lvnjx9wvb4nq-pw-viz-0.2.0.drv'.
```

**Comment by tobiasBora** [comment]
This seems to be somewhat related to [clang - Rust bindgen causes a "is not a valid Ident" error on build - Stack Overflow](https://stackoverflow.com/questions/76443280/rust-bindgen-causes-a-is-not-a-valid-ident-error-on-build) that points to this bug [thread 'main' panicked at '"__mbstate_t_(anonymous_struct_at_foo_cc_2_3)" is not a valid Ident' · Issue #2312 · rust-lang/rust-bindgen · GitHub](https://github.com/rust-lang/rust-bindgen/issues/2312) which is apparently due to this commit https://github.com/llvm/llvm-project/commit/19e984ef8f49bc3ccced15621989fa9703b2cd5b#diff-cde6fae42b53073a85e0901697f60ea80262705f86643b7b27524c7bd24c6f80.

The SO answer mentions:

>    Specifically, that commit was for Clang 16. Therefore, either downgrade Clang to a version before 16, or update your bindgen crate to v0.62.0 or above (which is when the fix was applied).

so it seems that somehow we need to update some rust stuff. Since it seems like this affect multiple packages, this may need to be done globally? Maybe the maintainers of rustc @cstrahan @globin @Havvy … know more?

**Comment by Atemu** [comment]
Could someone bisect this?

**Comment by colemickens** [comment]
I don't know if this is helpful, but I'm pretty sure I had to knock `zenith` out of my config for a very similar error as I saw with `pw-viz` on the same channel bump. Just in case it helps with the investigation. (Please re-confirm so I don't lead you down a wild goose chase)

**Comment by frantisekhanzlikbl** [comment]
I can confirm that zenith fails to build on current unstable with a similar error message.

<details>
<summary>build log</summary>

```console
error: failed to run custom build command for `linux-taskstats v0.2.0`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/linux-taskstats-5354d18bd6482380/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=wrapper.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/genetlink.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/types.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm/types.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm-generic/types.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm-generic/int-ll64.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm/bitsperlong.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm-generic/bitsperlong.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/posix_types.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/stddef.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm/posix_types.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm/posix_types_64.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm-generic/posix_types.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/asm/bitsperlong.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/netlink.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/const.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/socket.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/types.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/taskstats.h
  cargo:rerun-if-changed=/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2.38-27-dev/include/linux/types.h

  --- stderr
  thread 'main' panicked at /build/cargo-vendor-dir/proc-macro2-1.0.47/src/fallback.rs:756:9:
  "__kernel_sockaddr_storage_union_(anonymous_at_/nix/store/mrgib0s2ayr81xv1q84xsjg8ijybalq3-glibc-2_38-27-dev/include/linux/socket_h_17_2)" is not a valid Ident
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

</details>

**Comment by eclairevoyant** [comment]
#272090 for the zenith issue

**Comment by FliegendeWurst** [comment]
Fixed in #340974
